src/apps/relay/dbdrivers/dbd_sqlite.c

```diff
@@ -966,208 +991,211 @@ static void sqlite_auth_ping(void *rch) { UNUSED_ARG(rch); }
 
-        while (1) {
-          int res = sqlite3_step(st);
-          if (res == SQLITE_ROW) {
+    sqlite3_stmt *st = NULL;
+    if (sqlite3_prepare(sqliteconnection, statement, -1, &st, 0) == SQLITE_OK) {
 
-            char *oval = strdup((const char *)sqlite3_column_text(st, 0));
-            char *rval = strdup((const char *)sqlite3_column_text(st, 1));
+      ur_string_map *o_to_realm_new = ur_string_map_create(free);
 
-            get_realm(rval);
-            ur_string_map_value_type value = rval;
-            ur_string_map_put(o_to_realm_new, (ur_string_map_key_type)oval, value);
+      // TODO: Validate column count...
+      for (int stepResult = sqlite3_step(st); stepResult != SQLITE_DONE; stepResult = sqlite3_step(st)) {
 
-            free(oval);
+        if (stepResult == SQLITE_ROW) {
 
-          } else if (res == SQLITE_DONE) {
-            break;
-          } else {
-            const char *errmsg = sqlite3_errmsg(sqliteconnection);
-            TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Error retrieving SQLite DB information: %s\n", errmsg);
-            break;
-          }
-        }
+          // TODO: Why does oval need to be strdup? It seems used read-only, non-owning?
+          char *oval = strdup((const char *)sqlite3_column_text(st, 0));
+          char *rval = strdup((const char *)sqlite3_column_text(st, 1));
 
-        update_o_to_realm(o_to_realm_new);
+          get_realm(rval);
+          ur_string_map_value_type value = rval;
+          ur_string_map_put(o_to_realm_new, (ur_string_map_key_type)oval, value);
 
-      } else {
-        const char *errmsg = sqlite3_errmsg(sqliteconnection);
-        TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Error retrieving SQLite DB information: %s\n", errmsg);
+          free(oval);
+
+        } else {
+          const char *errmsg = sqlite3_errmsg(sqliteconnection);
+          TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Error retrieving SQLite DB information: %s\n", errmsg);
+          break;
+        }
       }
-      sqlite3_finalize(st);
 
-      sqlite_unlock(0);
+      update_o_to_realm(o_to_realm_new);
+
+    } else {
+      const char *errmsg = sqlite3_errmsg(sqliteconnection);
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Error retrieving SQLite DB information: %s\n", errmsg);
     }
 
-    {
-      {
-        size_t i = 0;
-        size_t rlsz = 0;
+    sqlite3_finalize(st);
 
-        lock_realms();
-        rlsz = realms_list->sz;
-        unlock_realms();
+    sqlite_unlock(0);
+  }
 
-        for (i = 0; i < rlsz; ++i) {
+  {
+    lock_realms();
+    size_t const rlsz = realms_list->sz;
+    unlock_realms();
 
-          char *realm = realms_list->secrets[i];
+    for (size_t i = 0; i < rlsz; ++i) {
+      char *realm = realms_list->secrets[i];
 
-          realm_params_t *rp = get_realm(realm);
+      realm_params_t *rp = get_realm(realm);
 
-          lock_realms();
-          rp->options.perf_options.max_bps = turn_params.max_bps;
-          unlock_realms();
+      lock_realms();
```

Comment:
This function was particularly gnarly.

Locking a mutex for a single variable, when there are 2 others to modify as well?

I don't fully know what this function is actually trying to accomplish, but it's a pretty good one for an expert to review.